great_expectations/expectations/core/expect_compound_columns_to_be_unique.py

```diff
@@ -96,13 +96,15 @@ def _atomic_prescriptive_template(
         template_str = (
             f"Values for given compound columns must be unique together{mostly_str}: "
         )
-        for idx in range(len(params["column_list"]) - 1):
-            template_str += "$column_list_" + str(idx) + ", "
-            params["column_list_" + str(idx)] = params["column_list"][idx]
-
-        last_idx = len(params["column_list"]) - 1
-        template_str += "$column_list_" + str(last_idx)
-        params["column_list_" + str(last_idx)] = params["column_list"][last_idx]
+        column_list = params.get("column_list") if params.get("column_list") else []
+        if len(column_list) > 0:
+            for idx in range(len(column_list) - 1):
+                template_str += "$column_list_" + str(idx) + ", "
+                params["column_list_" + str(idx)] = column_list[idx]
```

Comment:
```suggestion
        if len(column_list) > 0:
            for idx, val in enumerate(column_list):
                template_str += f"$column_list_{idx}, "
                params[f"column_list_{idx}"] = val
```

Just making some minor suggestions here:
1. Since we're using both indices and values, could we use `enumerate`?
2. This probably won't cause a noticeable performance penalty but the immutability of strings makes concatenation a pretty costly process. Instead of combining multiple strings together and then using `+=`, I proposed an f-string to cut down on some of that work. An additional approach would be one concatenation at the end (but I don't think this is entirely necessary):
```
foo = []
for idx, val in enumerate(column_list):
    foo.append(f"$column_list_{idx}, ")
    ...

template_str += ''.join(string for string in foo)
```

Reply:
Good catch, not sure why enumerate wasn't used there (I didn't write section, just indented it in new conditional). Will adjust.